src/k8s-extension/azext_k8s_extension/partner_extensions/BackupAgent.py

```diff
@@ -0,0 +1,81 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=unused-argument
+
+import datetime
+import json
+
+from ..utils import get_cluster_rp_api_version
+
+from knack.log import get_logger
+
+from azure.cli.core.azclierror import InvalidArgumentValueError
+from azure.cli.core.commands.client_factory import get_subscription_id
+from msrestazure.azure_exceptions import CloudError
+
+from ..vendored_sdks.models import Extension
+
+from .DefaultExtension import DefaultExtension
+
+from .._client_factory import (
+    cf_resources)
+
+logger = get_logger(__name__)
+
+
+class AzureBackupAgent(DefaultExtension):
+    def __init__(self):
+        # constants for configuration settings.
+        self.DEFAULT_RELEASE_NAMESPACE = 'backupagent-ns'
+
+    def Create(self, cmd, client, resource_group_name, cluster_name, name, cluster_type, extension_type,
+               scope, auto_upgrade_minor_version, release_train, version, target_namespace,
+               release_namespace, configuration_settings, configuration_protected_settings,
+               configuration_settings_file, configuration_protected_settings_file):
+
+        # set release name explicitly to backupagent-ns
+        target_namespace = self.DEFAULT_RELEASE_NAMESPACE
```

Comment:
Do you not want this to be user-overridable? Service should provide the default namespace but this will overwrite the user intention every time